mlir/lib/Dialect/MemRef/Transforms/EmulateNarrowType.cpp

```diff
@@ -335,7 +279,48 @@ struct ConvertMemRefReinterpretCast final
           op->getLoc(), "subview with rank > 1 is not supported");
     }
 
-    return convertCastingOp(rewriter, adaptor, op, newTy);
+    Type convertedElementType = newTy.getElementType();
```

Comment:
Could you avoid this inlining? Will help me understand the change better.

Reply:
sure, done. Now it is only used by the ReinterpretCastOp pattern, so I remove the template.